### PR TITLE
Add CLI call for UpdateActivityOptions API

### PR DESCRIPTION
**Pull request by ychebotarev**
## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,25 +11,25 @@ require (
...
 	github.com/stretchr/testify v1.9.0
 	github.com/temporalio/ui-server/v2 v2.32.0
-	go.temporal.io/api v1.38.0
-	go.temporal.io/sdk v1.29.1
-	go.temporal.io/server v1.25.2
-	google.golang.org/grpc v1.65.0
-	google.golang.org/protobuf v1.34.2
+	go.temporal.io/api v1.43.0
+	go.temporal.io/sdk v1.30.1
+	go.temporal.io/server v1.26.2-125.1
```
Comment (josh-berry) [inline go.mod:22]: We can't merge this in `main` without a released open-source server version. Please target your PR to `next-server` instead.

File: temporalcli/commands.activity_test.go [hunk 2]
```diff
@@ -106,6 +106,32 @@ func (s *SharedServerSuite) TestActivity_Fail_InvalidDetail() {
...
+		"--schedule-to-start-timeout", "5s",
+		"--start-to-close-timeout", "10s",
+		"--heartbeat-timeout", "20s",
+		"--retry-initial-interval", "5s",
+		"--retry-maximum-interval", "60s",
+		"--retry-backoff-coefficient", "2",
+		"--retry-maximum-attempts", "5",
+	)
+	// atm, the command is not enabled
+	s.Error(res.Err)
```
Comment (josh-berry) [inline temporalcli/commands.activity_test.go:132]: We'll need to be able to actually verify the behavior here before merge.

File: temporalcli/commandsgen/commands.yml [hunk 3]
```diff
@@ -204,10 +204,10 @@ commands:
           The command execution timeout. 0s means no timeout.
 
   - name: temporal activity
-    summary: Complete or fail an Activity
+    summary: Complete, update or fail an Activity
     description: |
-      Update an Activity's state to completed or failed. This marks an Activity
-      as successfully finished or as having encountered an error:
+      Update an Activity's options or update an Activity's state to completed or failed.
+      Updating activity state marks an Activity as successfully finished or as having encountered an error.
```
Comment (josh-berry) [inline temporalcli/commandsgen/commands.yml:210]: Please wrap to 80 columns per the "Wrapping" section of the style guide at the top of this file. Same goes for the options and command descriptions.

File: temporalcli/commandsgen/commands.yml [hunk 4]
```diff
@@ -287,6 +288,81 @@ commands:
...
+          --workflow-id YourWorkflowId \
+          --task-queue NewTaskQueueName \
+          --schedule-to-close-timeout DURATION\
+          --schedule-to-start-timeout DURATION\
+          --start-to-close-timeout DURATION\
+          --heartbeat-timeout DURATION\
+          --retry-initial-interval DURATION\
+          --retry-maximum-interval DURATION\
+          --retry-backoff-coefficient NewBackoffCoefficient\
+          --retry-maximum-attempts NewMaximumAttempts"
```
Comment (josh-berry) [inline temporalcli/commandsgen/commands.yml:310]: Inconsistent use of whitespace in your `\`s, and you have a stray `"`

File: go.mod (hunk 1 above)
Comment (cretz) [inline go.mod:22]: We cannot update `main` to a non-stable tag. But there is the `next-server` branch you can change this PR to target

File: temporalcli/commandsgen/commands.yml [hunk 5]
```diff
@@ -287,6 +288,81 @@ commands:
     option-sets:
       - workflow reference
 
+
+  - name: temporal activity update
```
Comment (cretz) [inline temporalcli/commandsgen/commands.yml:292]:
```suggestion
  - name: temporal activity update-options
```

Can we change this to `update-options`? I am afraid of a future concept of an "activity update" that is unrelated to updating options

File: temporalcli/commands.activity_test.go [hunk 6]
```diff
@@ -106,6 +106,32 @@ func (s *SharedServerSuite) TestActivity_Fail_InvalidDetail() {
 	s.Nil(failed)
 }
 
+func (s *SharedServerSuite) TestActivityOptionsUpdate_Accept() {
```
Comment (cretz) [inline temporalcli/commands.activity_test.go:109]: Can we add one more test confirming a partial update doesn't mutate the existing value?

File: temporalcli/commandsgen/commands.yml [hunk 7]
```diff
@@ -287,6 +288,81 @@ commands:
...
+        description: Activity ID.
+        required: true
+      - name: task-queue
+        type: string
+        description: Name of the task queue for the Activity.
+      - name: schedule-to-close-timeout
+        type: duration
+        description: |
+          Indicates how long the caller is willing to wait for an activity completion. 
+          Limits how long retries will be attempted.
```
Comment (josh-berry) [inline temporalcli/commandsgen/commands.yml:325]: For your option descriptions, please follow the grammar recommendations in the style guide at the top of the file. You can look at other options (e.g. the workflow-start timeout options) if you need examples.

File: temporalcli/commands.activity_test.go [hunk 8]
```diff
@@ -106,6 +106,32 @@ func (s *SharedServerSuite) TestActivity_Fail_InvalidDetail() {
...
+		"--schedule-to-close-timeout", "60s",
+		"--schedule-to-start-timeout", "5s",
+		"--start-to-close-timeout", "10s",
+		"--heartbeat-timeout", "20s",
+		"--retry-initial-interval", "5s",
+		"--retry-maximum-interval", "60s",
+		"--retry-backoff-coefficient", "2",
+		"--retry-maximum-attempts", "5",
+	)
+	// atm, the command is not enabled
```
Comment (cretz) [inline temporalcli/commands.activity_test.go:131]: Should have PR wait to be merged until test can confirm behavior

File: temporalcli/commandsgen/commands.yml [hunk 9]
```diff
@@ -287,6 +288,81 @@ commands:
...
+      - name: schedule-to-start-timeout
+        type: duration
+        description: |
+          Limits time an activity task can stay in a task queue before a worker picks it up. 
+          This timeout is always non retryable, as all a retry would achieve is to put it back into the same
+          queue. Defaults to `schedule_to_close_timeout` or workflow execution timeout if not specified.
```
Comment (josh-berry) [inline temporalcli/commandsgen/commands.yml:331]: Avoid backticks unless you are intending to specify the metavariable for this option (see style guide).

```suggestion
          queue. Defaults to the schedule-to-close timeout or workflow execution timeout if not specified.
```

**Review by josh-berry — COMMENTED** [review]
I just did a quick pass right before the end of my workday; other SDK folks may have additional comments.

File: temporalcli/commands.activity_test.go (hunk 6 above)
Comment (ychebotarev) [reply temporalcli/commands.activity_test.go:109]: what are we testing with that? If server implementation - it is already tested.

Comment (cretz) [reply temporalcli/commands.activity_test.go:109]: Just an extra confirmation. Many (most?) of our CLI tests are end to end tests with the server implementation. We don't have to test everything, but I would like to confirm that only setting some CLI flags leaves other values intact.

Comment (ychebotarev) [reply temporalcli/commands.activity_test.go:109]: ok

File: temporalcli/commands.activity_test.go (hunk 8 above)
Comment (ychebotarev) [reply temporalcli/commands.activity_test.go:131]: ok looks like we can change dynamic config values, so the output can be tested

**Comment by ychebotarev** [comment]
create new PR for next-server

**Pull request closed without merging**